change sparse COO comparison strategy in assert_close

pmeier opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69751
* #67796
* #70304
* #69750
* #68911
* __->__ #68728
* #68802
* #68977

This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc @nikitaved @pearu @cpuhrsch @IvanYashchuk

Differential Revision: [D33542996](https://our.internmc.facebook.com/intern/diff/D33542996)

pmeier committed:
change sparse COO comparison strategy in assert_close

[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

IvanYashchuk reviewed (approved):
I think it's a good idea to sum the duplicates explicitly in the test before calling assert equal.

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

pmeier committed:
Update on "change sparse COO comparison strategy in assert_close"


This removes the ability for `assert_close` to `.coalesce()` the tensors internally. Additionally, we now also check `.sparse_dim()`. Sparse team: please make sure that is the behavior you want for all sparse COO comparisons in the future. #67796 will temporarily keep BC by always coalescing, but in the future `TestCase.assertEqual` will no longer do that.

cc nikitaved pearu cpuhrsch @IvanYashchuk

[ghstack-poisoned]

mruberry commented on torch/testing/_comparison.py:
```diff
@@ -50,6 +50,14 @@ def to_error(self) -> Exception:
     torch.complex64: (1.3e-6, 1e-5),
     torch.complex128: (1e-7, 1e-7),
 }
+# The default tolerances of torch.float32 are used for quantized dtypes, because quantized tensors are compared in
```
Nice comment

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33542996).
